src/main/java/com/google/api/codegen/transformer/java/JavaMethodViewGenerator.java

```diff
@@ -0,0 +1,147 @@
+/* Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.api.codegen.transformer.java;
+
+import com.google.api.codegen.config.FieldConfig;
+import com.google.api.codegen.config.FlatteningConfig;
+import com.google.api.codegen.config.MethodConfig;
+import com.google.api.codegen.config.MethodModel;
+import com.google.api.codegen.config.SampleSpec.SampleType;
+import com.google.api.codegen.transformer.ImportTypeTable;
+import com.google.api.codegen.transformer.InterfaceContext;
+import com.google.api.codegen.transformer.MethodContext;
+import com.google.api.codegen.transformer.StaticLangApiMethodTransformer;
+import com.google.api.codegen.viewmodel.StaticLangApiMethodView;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Contains the common logic for generating view models for GAPIC surface methods. This is used in
+ * the generation of both the client libraries and of the standalone samples for each of the library
+ * surface methods. Since in either case the resulting methods have one or more samples, this
+ * populates the view models with appropriate sample view models.
+ */
+public class JavaMethodViewGenerator {
+
+  final StaticLangApiMethodTransformer clientMethodTransformer;
+
+  public JavaMethodViewGenerator(SampleType sampleType) {
+    clientMethodTransformer = new StaticLangApiMethodTransformer(sampleType);
+  }
+
+  /**
+   * Generates the StaticLangApiMethodView for each of the methods in the InterfaceContext
```

Comment:
End sentence with a period.

Reply:
Done.  What's our convention for Java doc? It seems like in some cases we don't.

Reply:
Convention: Add javadoc when it seems like an explanation is necessary.

When we decide we want more out-of-team adoption down the road, we'll have to start being more thorough.

Reply:
_That_ part I got (even if in some cases I disagreed with the amount of information previously provided) and is what I've been doing: javadoc where it seems illuminating.

I meant my question more superficially: what is the convention about end-of-line punctuation in javadoc?

Reply:
The period drives javadoc logic - it defines the end of the short-form comment displayed in summary tables. http://www.oracle.com/technetwork/java/javase/tech/index-137868.html (search for "period")

Reply:
Cool, thanks, I was looking for something like that. It seems we don't always follow that, but I'll use that as my style guide in this repo.